source/MaterialXCore/Definition.cpp

```diff
@@ -186,7 +186,7 @@ StringVec TargetDef::getMatchingTargets() const
 vector<UnitDefPtr> UnitTypeDef::getUnitDefs() const
 {
     vector<UnitDefPtr> unitDefs;
-    for (UnitDefPtr unitDef : getDocument()->getChildrenOfType<UnitDef>())
+    for (UnitDefPtr unitDef : getDocument()->getChildrenOfType<UnitDef>(EMPTY_STRING, getDocument()->getDataLibrary()))
```

Comment:
@jstone-lucasfilm IIRC there is a cyclic dependency between `Element` and `Document` 
both getChildOfType and getChildrenOfType use the passed datalibrary as a ConstElementPtr 
In order to implement what you propose I think we might need to move datalibrary into the Element

Reply:
@ashwinbhat Good point, and another option would be to add overrides to the `Document` class for `getChildOfType` and `getChildrenOfType`, and only these overrides would check the associated data library and search through it for matching children.

That might be a simpler approach for readers to understand and maintain, and I believe it should be fine to remove both the Python and JavaScript bindings for `getChildOfType` and `getChildrenOfType` for now, so that we don't need to solve these at the same time as the C++ API.